arden/ArdenLexer.g4

```diff
@@ -0,0 +1,344 @@
+TRUTHVALUE:     'truth value';
+TYPE:           'type:';
+TYPE_CODE:      'data'[_-]'driven';
+UNTIL:          'until';
+UPPERCASE:      'uppercase';
+URGENCY:        'urgency:';
+USING:          'using';
+VALIDATION:     'validation:';
+VALIDATION_CODE:'production' | 'research' | 'testing' | 'expired';
+VARIABLE:       'variable';
+VARIANCE:       'variance';
+VERSION:        'version:' -> mode(TextMode);
+WEEK:           'week' | 'weeks';
+WEEKDAYLITERAL: 'sunday' | 'monday' | 'tuesday' | 'wednesday' | 'thursday' | 'friday' | 'saturday';
+WHERE:          'where';
+WHILE:          'while';
+WITH:           'with';
+WITHIN:         'within';
+WRITE:          'write';
+YEAR:           'year' | 'years';
+
+// Seperators
+LPAREN:         '(';
+RPAREN:         ')';
+LBRACE:         '{' -> mode(DataMapping);
+RBRACE:         '}';
+LBRACK:         '[';
+RBRACK:         ']';
+SC:             ';';
+DSC:            ';;';
+COLON:          ':';
+DOT:            '.';
+COMMA:          ',';
+
+// Operators
+ASSIGN:         ':=';
+PLUS:           '+';
+MINUS:          '-';
+MUL:            '*';
+DIV:            '/';
+POWER:          '**';
+EQ:             '=' | 'eq';
+LT:             '<' | 'lt';
+GT:             '>' | 'gt';
+LE:             '<=' | 'le';
+GE:             '>=' | 'ge';
+NE:             '<>' | 'ne';
+DOR:            '||';
+
+// Digit constructs
+NUMBER
+    :           Digit+ DOT? Digit* Exponent
+    |           DOT Digit+ Exponent
+    ;
+
+// Date constructs
+TIMEOFDAY:      Digit Digit COLON Digit Digit Seconds TimeZone;
+
+ISO_DATE:       Digit Digit Digit Digit MINUS Digit Digit MINUS Digit Digit;
+
+ISO_DATE_TIME:  Digit Digit Digit Digit MINUS Digit Digit MINUS Digit Digit 't' Digit Digit COLON Digit Digit COLON Digit Digit FractionalDigit TimeZone;
+
+// String constructs
+CITATION:       Digit+ DOT ' ' ('support' | 'refute')?;
+
+TERM:           '\''  .*? '\'';
+
+STRING:         '"' ( '""' | ~["] )* '"';
+
+// Creates an identifier with max of 80 chars
+IDENTIFIER
+    :           StartIdentifier RestIdentifier RestIdentifier RestIdentifier RestIdentifier RestIdentifier RestIdentifier RestIdentifier
+    ;
+
+// Comment
+COMMENT:        '/*' .*? '*/' -> channel(HIDDEN);
+
+LINE_COMMENT:   '//' ~[\r\n]* -> channel(HIDDEN);
+
+// Whitespace
+WS:             [ \t\r\n]+ -> channel(HIDDEN);
+
+// Fragment rules
+// Digit fragments
+fragment Digit
+    :           [0-9]
+    ;
+
+fragment FractionalDigit
+    :          (DOT Digit+)?
+    ;
+
+fragment Exponent
+    :           ('e' [+-]? Digit+)?
+    ;
+
+// Date fragments
+fragment Seconds
+    :           (COLON Digit Digit FractionalDigit)?
+    ;
+
+fragment TimeZone
+    :           'z'?
+    |           (PLUS | MINUS) Digit Digit COLON Digit Digit
+    ;
+
+// String fragments
+fragment Letter
+    :           [a-z]
+    ;
+
+// Start of the identifier 10 chars, starting with a letter
+fragment StartIdentifier
+    :           Letter ID? ID? ID? ID? ID? ID? ID? ID? ID?
```

Comment:
Did you mean the following? It accepts a sequence from 0 to 9 `ID` elements:

```suggestion
    :           Letter (ID (ID (ID (ID (ID (ID (ID (ID ID?)?)?)?)?)?)?)?)?
```

The same applicable to other similar rules defined below (`RestIdentifier`, `MlmIDStart`, `MlmIDRest`).

Reply:
Yes this is exactly what I mean.